### DISCUSSION TITLE
I encountered a problem in the chapter 'Elasticities' in the MICOM documentation.

### DISCUSSION THREAD

**Discussion by Nagi-Kazami** [micom-dev/micom#107] [Q&A] [answered]
Hi, I am a new user of MICOM.
I was going through the training in the documentation and encountered a problem in the chapter 'Elasticities'.
The output is very different from the sample. 
Why?

The table below shows the output of elasticities() in my environment.

reaction	taxon	effector	direction	elasticity	type
0	EX_ac_m	medium	EX_akg_m	forward	0.0	exchanges
1	EX_acald_m	medium	EX_akg_m	forward	0.0	exchanges
2	EX_akg_m	medium	EX_akg_m	reverse	0.0	exchanges
3	EX_co2_m	medium	EX_akg_m	forward	0.0	exchanges
4	EX_etoh_m	medium	EX_akg_m	forward	0.0	exchanges

reaction	taxon	effector	direction	elasticity	type
60	EX_ac_m	medium	EX_glc__D_m	forward	-0.280126	exchanges
61	EX_acald_m	medium	EX_glc__D_m	reverse	-6.899914	exchanges
62	EX_akg_m	medium	EX_glc__D_m	reverse	5.149733	exchanges
63	EX_co2_m	medium	EX_glc__D_m	forward	0.929889	exchanges
64	EX_etoh_m	medium	EX_glc__D_m	forward	-5.989809	exchanges
65	EX_for_m	medium	EX_glc__D_m	forward	-0.606646	exchanges
66	EX_fru_m	medium	EX_glc__D_m	reverse	5.138715	exchanges
67	EX_fum_m	medium	EX_glc__D_m	reverse	6.576951	exchanges
68	EX_glc__D_m	medium	EX_glc__D_m	reverse	1.002646	exchanges
69	EX_gln__L_m	medium	EX_glc__D_m	reverse	3.414139	exchanges
70	EX_glu__L_m	medium	EX_glc__D_m	reverse	4.248035	exchanges
71	EX_h_m	medium	EX_glc__D_m	forward	1.048605	exchanges
72	EX_h2o_m	medium	EX_glc__D_m	forward	0.955087	exchanges
73	EX_lac__D_m	medium	EX_glc__D_m	forward	-5.589720	exchanges
74	EX_mal__L_m	medium	EX_glc__D_m	reverse	10.140689	exchanges
75	EX_nh4_m	medium	EX_glc__D_m	reverse	1.049425	exchanges
76	EX_o2_m	medium	EX_glc__D_m	reverse	0.923621	exchanges
77	EX_pi_m	medium	EX_glc__D_m	reverse	1.052104	exchanges
78	EX_pyr_m	medium	EX_glc__D_m	forward	-8.369191	exchanges
79	EX_succ_m	medium	EX_glc__D_m	reverse	10.330858	exchanges

**Comment by cdiener** [comment, marked as answer]
Yeah this is because the elasticities are very dependent on the initial solution and the one in the docs were generated with CPLEX and not with OSQP (which gives the results you have seen). I'm planning on rewriting the elasticities part soon, so hopefully that will fix this.
